src/core/qfyaml_mod.F90

```diff
@@ -596,6 +642,216 @@ SUBROUTINE QFYAML_Read_File( yml, fileName, yml_anchored, RC )
+             var_name   = yml_anchored%vars(N)%var_name
+
+             ! Find all target variables with the given value of anchor_ptr,
+             ! and return the start and ending indices in the yml config object.
+             CALL Get_Anchor_Info( yml        = yml,                                &
+                                   anchor_ptr = anchor_ptr,                         &
+                                   begin_ix   = begin_ix,                           &
+                                   end_ix     = end_ix,                             &
+                                   anchor_cat = anchor_cat )
+
+             ! Loop over all target variables containing the value of anchor_ptr
+             DO anchor_ix = begin_ix, end_ix
+
+                ! Variable with the anchor
+                var_w_anchor = yml%vars(anchor_ix)%var_name
+
+                ! Variable that we want to point to the anchor
+                I = INDEX( var_w_anchor, QFYAML_category_separator )
+                var_pt_to_anchor = TRIM( category )                             // &
+                                   QFYAML_category_separator                    // &
+                                   var_w_anchor(I+1:)
+
+                ! Create a new variable for this category,
+                ! copying the fields of the variable with the anchor.
+                CALL Copy_Anchor_Variable( yml              = yml,                 &
+                                           anchor_ix        = anchor_ix,           &
+                                           var_w_anchor     = var_w_anchor,        &
+                                           var_pt_to_anchor = var_pt_to_anchor,    &
+                                           RC               = RC                  )
+
+                ! Trap potential errors
+                IF ( RC /= QFYAML_Success ) THEN
+                   errMsg = 'Error encountered in "Copy_Anchor_Variable"!'
+                   CALL Handle_Error( errMsg, RC, thisLoc )
+                   RETURN
+                ENDIF
+             ENDDO
+          ENDDO
+
+          ALLOCATE(names(0), STAT=RC)
+         !  IF ( RC /= 0 ) THEN
+         !     errMsg = 'Error allocating "names"!'
+         !     CALL Handle_Error( errMsg, RC, thisLoc )
+         !     RETURN
+         !  ENDIF
+          current = ""
+          i = 0
+          do n = 1, yml%num_vars
+            if (len(TRIM(yml%vars(n)%category)) > 0) then
+               if ( i == 0 ) then
+                  i = i + 1
+                  current = TRIM(yml%vars(n)%category)
+                  names = [names, yml%vars(n)%category]
+               else if (TRIM(yml%vars(n)%category) /= TRIM(current)) then
+                  i = i + 1
+                  names = [names, yml%vars(n)%category]
+                  current = TRIM(yml%vars(n)%category)
+               endif
+            endif
+         enddo
+
+         ALLOCATE(species_names(i), STAT=RC)
```

Comment:
`names` should have all the names at the end of the previous block? This block seems to just be a duplicate.